test/common/index.js

```diff
@@ -746,3 +749,47 @@ exports.getTTYfd = function getTTYfd() {
   }
   return tty_fd;
 };
+
+// exports.tag*: tag functions for tagged template literals
+
+// Removes all literal line breaks and leading spaces
+exports.tagGlue = function tagGlue(strings, ...expressions) {
+  const breaks = /\n */g;
+
+  return expressions.reduce(
+    (acc, expr, i) => `${acc}${expr}${strings[i + 1].replace(breaks, '')}`,
+    strings[0].replace(breaks, '')
+  );
+};
+
+// Replaces all literal line breaks and leading spaces with a space, then trims
+exports.tagUnwrap = function tagUnwrap(strings, ...expressions) {
+  const breaks = /(\n *)+/g;
+
+  return expressions.reduce(
+    (acc, expr, i) => `${acc}${expr}${strings[i + 1].replace(breaks, ' ')}`,
+    strings[0].replace(breaks, ' ')
+  ).replace(/^ | $/g, '');
+};
+
+// Removes two framing line breaks and only auxiliary indents
+exports.tagLF = function tagLF(strings, ...expressions) {
+  const [, firstIndent = ''] = strings[0].match(/\n+( *)/) || [];
```

Comment:
@joyeecheung would fail for (badly formated) literals like:
```js
const tagged = tagLF`
<!doctype html>
<html>
  <head>
    <meta charset="UTF-8">
    <title></title>
  </head>
  <body>
    ${process.versions}
  </body>
</html>
`;
```
would give `firstIndent === undefined`